substrait_consumer/producers.py

```diff
@@ -138,6 +144,83 @@ def name(self):
         return "IsthmusProducer"
 
 
+class DataFusionProducer:
+    def __init__(self, db_connection=None):
+        self.ctx = SessionContext()
+        if db_connection is not None:
+            self.db_connection = db_connection
```

Comment:
if `self._` approach is adopted we may need reflect that change to similar attributes.